### PR TITLE
Bookstore

### PR DESCRIPTION
**Pull request by drozddmitriy**

### CODE REVIEW

File: app/helpers/books_helper.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+module BooksHelper
```
Comment (dzemlianoi-double) [inline app/helpers/books_helper.rb]: remove

File: app/models/author.rb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+class Author < ApplicationRecord
+  has_and_belongs_to_many :books
```
Comment (dzemlianoi-double) [inline app/models/author.rb]: better to use has_many 📚 , through: :book_authors

File: app/views/books/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+  <h1 class="general-title-margin">Сatalog</h1>
+  <div class="hidden-xs clearfix">
+    <div class="dropdowns pull-right">
+      <div class="dropdown width-240">
+        <a href="#" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false" class="dropdown-toggle lead small">Newest first
+          <i class="fa fa-angle-down dropdown-icon pull-right"></i>
+        </a>
+        <ul class="dropdown-menu">
+          <li>
+            <a href="#">Newest first</a>
```
Comment (dzemlianoi-double) [inline app/views/books/index.html.erb]: i18n

File: db/migrate/20190826214617_add_details_to_books.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class AddDetailsToBooks < ActiveRecord::Migration[5.2]
+  def change
+    add_column :books, :category_id, :integer
```
Comment (dzemlianoi-double) [inline db/migrate/20190826214617_add_details_to_books.rb]: index

File: test/fixtures/categories.yml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# Read about fixtures at http://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
Comment (dzemlianoi-double) [inline test/fixtures/categories.yml]: remove test folder

File: app/controllers/books_controller.rb [hunk 6]
```diff
@@ -0,0 +1,12 @@
+class BooksController < ApplicationController
+  def index
+    @categories = Category.all
+
+    @books = Book.first(8)
+  end
+
+  def show
+    @book = Book.find_by(id: params[:id])
+    # binding.pry
```
Comment (za4emnik) [inline app/controllers/books_controller.rb]: ?

File: app/controllers/books_controller.rb [hunk 7]
```diff
@@ -0,0 +1,12 @@
+class BooksController < ApplicationController
+  def index
+    @categories = Category.all
+
+    @books = Book.first(8)
```
Comment (za4emnik) [inline app/controllers/books_controller.rb]: 8 - const

File: app/mailers/application_mailer.rb [hunk 8]
```diff
@@ -0,0 +1,4 @@
+class ApplicationMailer < ActionMailer::Base
+  default from: 'from@example.com'
```
Comment (za4emnik) [inline app/mailers/application_mailer.rb]: `from@example.com` move to ENV

File: app/views/books/index.html.erb [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+          </li>
+          <li>
+            <a href="#">Popular first</a>
```
Comment (za4emnik) [inline app/views/books/index.html.erb]: move all text to I18n

File: app/views/books/index.html.erb [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+          <li>
+            <a href="#">Price: Hight to low</a>
+          </li>
+        </ul>
+      </div>
+    </div>
+    <ul class="list-inline pt-10 mb-25 mr-240">
+      <li class="mr-35">
+        <a href="<%= books_path %>" class="filter-link"><%= I18n.t('views.books.all') %>
+          <span class="badge general-badge"><%= Book.all.count %></span></a>
```
Comment (za4emnik) [inline app/views/books/index.html.erb]: `Book.count`

File: config/database.yml [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+  database: bookstore_development
+
+test:
+  <<: *default
+  database: bookstore_test
+
+production:
+  <<: *default
+  database: bookstore_production
+  username: bookstore
```
Comment (za4emnik) [inline config/database.yml]: ENV

File: config/initializers/application_controller_renderer.rb [hunk 12]
```diff
@@ -0,0 +1,8 @@
+# Be sure to restart your server when you modify this file.
```
Comment (za4emnik) [inline config/initializers/application_controller_renderer.rb]: ?

File: test/application_system_test_case.rb [hunk 13]
```diff
@@ -0,0 +1,5 @@
+require "test_helper"
```
Comment (za4emnik) [inline test/application_system_test_case.rb]: remove this folder

File: app/views/books/index.html.erb [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+                  </a>
+                </div>
+              </div>
+            </div>
+          </div>
+          <div class="general-thumb-info">
+            <p class="title"><%= book.title %></p>
+            <p class="font-16 in-gold-500"><%= book.price %></p>
+            <span class="divider-sm"></span>
+            <p class="lead small"><%= book.authors.map { |author| author.firstname }.join(", ") %></p>
```
Comment (za4emnik) [inline app/views/books/index.html.erb]: move

File: app/views/books/show.html.erb [hunk 15]
```diff
@@ -0,0 +1,151 @@
...
+  <a href="<%= books_path %>" class="general-back-link">
+    <i class="fa fa-long-arrow-left mr-15"></i><%= I18n.t('views.books.results') %></a>
+  <div class="row">
+    <div class="col-sm-6">
+      <div class="product-gallery">
+        <div class="mb-20">
+          <img src="https://www.bramstein.com/static/smashing-book-5-photo-stack.9d5d114e3e.jpg" class="img-responsive"/></div>
+        <div class="images clearfix">
+          <a href="#" class="img-link">
+            <img src="https://www.bramstein.com/static/smashing-book-5-photo-stack.9d5d114e3e.jpg"/></a>
```
Comment (za4emnik) [inline app/views/books/show.html.erb]: it's a temporary solution?

File: app/views/books/show.html.erb [hunk 16]
```diff
@@ -0,0 +1,151 @@
...
+      </div>
+    </div>
+    <div class="col-sm-6">
+      <h1 class="mt-res-0"><%= @book.title %></h1>
+      <p class="lead small">
+        <%= BookDecorator.new(@book).author_full_name %>
+      </p>
+      <div class="row">
+        <div class="col-xs-12 col-md-4">
+          <p class="h1 in-gold-500 mt-10"><%= number_to_currency(@book.price, unit: "€") %></p>
```
Comment (za4emnik) [inline app/views/books/show.html.erb]: for what `unit: "€"`?

File: app/views/devise/unlocks/new.html.erb [hunk 17]
```diff
@@ -0,0 +1,16 @@
+<h2>Resend unlock instructions</h2>
```
Comment (za4emnik) [inline app/views/devise/unlocks/new.html.erb]: do you need this view?

File: config/locales/simple_form.en.yml [hunk 18]
```diff
@@ -0,0 +1,9 @@
+en:
+  simple_form:
+    "yes": 'Yes'
```
Comment (za4emnik) [inline config/locales/simple_form.en.yml]: for what `" "`?

File: spec/controllers/addresses_controller_spec.rb [hunk 19]
```diff
@@ -0,0 +1,50 @@
...
+    end
+
+    it 'assigns @addresses' do
+      expect(assigns(:addresses)).to match_array(user.addresses)
+    end
+  end
+
+  describe 'PUT #update' do
+    let(:address) { create(:address) }
+    let(:address_params) do
```
Comment (za4emnik) [inline spec/controllers/addresses_controller_spec.rb]: one line

File: spec/controllers/addresses_controller_spec.rb [hunk 20]
```diff
@@ -0,0 +1,50 @@
...
+      { address: attributes_for(:address) }
+    end
+
+    before { post :create, params: address_params }
+
+    it 'return redirect response' do
+      expect(response).to have_http_status(302)
+    end
+
+    it 'assigns @address' do
```
Comment (za4emnik) [inline spec/controllers/addresses_controller_spec.rb]: you need to check created object

File: spec/factories/addresses.rb [hunk 21]
```diff
@@ -0,0 +1,13 @@
+FactoryBot.define do
+  factory :address do |address|
+    firstname { FFaker::Name.first_name }
+    lastname { FFaker::Name.last_name }
+    address { FFaker::Address.street_address}
+    city { FFaker::Address.city }
+    zip { FFaker::AddressDE.zip_code }
+    country { FFaker::Address.country_code }
+    phone { FFaker::PhoneNumber.phone_number }
+    cast { 'billing' }
```
Comment (za4emnik) [inline spec/factories/addresses.rb]: trait

**Comment by za4emnik** [comment]
Check with rubocop, please

**Pull request merged**